dvc/remote/gdrive.py

```diff
@@ -73,6 +73,10 @@ def __init__(self, repo, config):
         super(RemoteGDrive, self).__init__(repo, config)
         self.no_traverse = False
         self.path_info = self.path_cls(config[Config.SECTION_REMOTE_URL])
+        self.bucket = (
+            config[Config.SECTION_REMOTE_URL].split("://")[1].split("/")[0]
```

Comment:
@efiop @MaxRis do we have a check somewhere that REMOTE_URL is an actual URL? We need to avoid Index error on `split('://')[1]` even if user puts a wrong URL into it. I suppose it should be somewhere where we determine the remote type by schema, right? Is that one enough to avoid the error here?

Reply:
also, just curious - do they actually call the root folder a `bucket`?

Reply:
@shcheklein you're right, ` gdrive:test` causes index error. It seems, if URL starts from `gdrive` the instance of RemoteGDrive is instantiated and no additional schema validation is done.


> also, just curious - do they actually call the root folder a bucket?

No, Google Drive doesn't have such terminology as "bucket". "bucket" is used as DVC's specific terminology ( "bucket" is used by others remotes ).

Reply:
how do the call "root" in the URL? directory? let's stick with GDrive termniniology. If it's a directory or folder - let's call it `root_folder` or something.

> `starts from gdrive`

so, does it mean that a "local" remote with a path `gdrive/test` will be considered a GDrive as well?

> no additional schema validation is done.

I wonder where should we fix this? per remote, or somewhere globally in the remote type detection code for example? what do you think @efiop ?

Reply:
> how do the call "root" in the URL? directory? let's stick with GDrive termniniology. If it's a directory or folder - let's call it root_folder or something.

There is, seems, no special naming. `root_folder` sounds good, but it might sounds similar to already available `root_id` variable, which is id of actual root directory, but not top most.

> so, does it mean that a "local" remote with a path gdrive/test will be considered a GDrive as well?

Sorry, it seems I was wrong, `gdrive/test` is recognized as local path. But still makes sense to add more validation ( `gdrive:test` is passed to remote without additional validation ).

Reply:
> ... already available root_id variable, which is id of actual root directory, but not top most.

I saw that you were using "root" before you changed to the `self.bucket`. Is that "root" special name that has `root_id` your are referring to? May be I don't then understand the logic at all. If a shared URL is just a regular one, just with a strange name (id instead of a human readable one), why do we need to change anything at all. Sorry, that I'm asking this. Probably you have answered this already somewhere.

Reply:
> Is that "root" special name that has root_id your are referring to?

"root" is the alias used by Google Drive to point to the user's top most directory. DVC references top most directory just by specifying "root" instead of unique id.

`root_id` is unique id of directory with DVC files.

Usually, it will be different directories.

> . If a shared URL is just a regular one, just with a strange name (id instead of a human readable one), why do we need to change anything at all

Following change https://github.com/iterative/dvc/pull/3016/files#diff-21bfd10d01c6715fc8634644730fa720R272 allows usage of ids as top directory. Currently usage of id as top directory in URL is broken. And this also the blocker for shared team drives, since the top directory always is referenced by id for them.

Reply:
It might makes sense to rename `root_id` to `dvc_root_id`, or something better.

Reply:
thanks, @MaxRis - my understanding is getting better, but I still have some questions. 

yes, `root_id` should be `remote_root_id`. Though, it's already part of the remote class. May be a comment would be enough to understand the difference.

So, does it mean that, before these URLs with Ids were introduced, all URLs always looked like:

`gdrive://root/path/to/remote`?

Reply:
b4a663a#diff-21bfd10d01c6715fc8634644730fa720R82

I believe it shouldn't fail under any input with index error or regular expression group not found error.


> So, does it mean that, before these URLs with Ids were introduced, all URLs always looked like:

gdrive://root/path/to/remote

I've tested such URLs as gdrive://ID` or gdrive://ED/path/to/remote with my initial changes, but it seems support was broken in finally merged PR after huge amount of modifications. My intention is to cover different URLs types with tests (hopefully, it will be not hard to verify that directories are created property at remote).

Reply:
ok, got it. Thanks.